src/Sarif/Comparers/ComparerHelpers.cs

```diff
@@ -0,0 +1,205 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Comparers
+{
+    internal class ComparerHelper
```

Comment:
>ComparerHelper [](http://example.com/codeflow?start=19&length=14)

Perhaps create class named ComparerExtensions.cs?

with extension methods.

public static bool TryCompare(this object left, object right, out int result)

Reply:
Changed from helper methods to extension methods. 
Renamed this method to 'TryReferenceCompares', following .net "ReferenceEquals" naming